controlplane/src/core/repositories/analytics/MetricsDashboardRepository.ts

```diff
@@ -178,215 +127,392 @@ export class MetricsDashboardRepository {
+      GROUP BY timestamp
+      ORDER BY timestamp ASC WITH FILL FROM 
+        toStartOfInterval(toDateTime('${start}'), INTERVAL ${granule} MINUTE)
+      TO
+        toDateTime('${end}')
+      STEP INTERVAL ${granule} minute
+    `);
+    };
 
-    const prevSeries = this.client.queryRange({
-      query: `sum(rate(cosmo_router_http_requests_total{${this.getQueryLabels(params)}}[${prange}]) * 60)`,
-      ...this.getRangeQueryProps({
-        range,
-        endDate: endDate - range * 60 * 60 * 1000,
-      }),
-    });
+    const series = querySeries(start, end);
+    const prevSeries = querySeries(prevStart, prevEnd);
 
-    const responses = await this.getResponses(rate, prevRate, top5, series, prevSeries);
-    const [medianResponse, prevMedianResponse, top5Response, seriesResponse, prevSeriesResponse] = responses;
+    const [medianResponse, prevMedianResponse, top5Response, seriesResponse, prevSeriesResponse] = await Promise.all([
+      medianRate,
+      prevMedianRate,
+      top5,
+      series,
+      prevSeries,
+    ]);
 
     return {
       data: {
-        value: parseValue(medianResponse.data?.result[0]?.value[1]),
-        previousValue: parseValue(prevMedianResponse.data?.result[0]?.value[1]),
-        top:
-          top5Response.data?.result?.map((v: any) => ({
-            name: v.metric.wg_operation_name || 'unknown',
-            value: parseValue(v.value[1]),
-          })) || [],
-        series: this.mapSeries(
-          endDate,
-          range,
-          seriesResponse.data?.result[0]?.values,
-          prevSeriesResponse.data?.result[0]?.values,
-        ),
+        value: parseValue(medianResponse[0]?.medianRate),
+        previousValue: parseValue(prevMedianResponse[0]?.medianRate),
+        top: top5Response.map((v) => ({
+          name: v.name,
+          value: parseValue(v.value),
+        })),
+        series: this.mapSeries(range, seriesResponse, prevSeriesResponse),
       },
-      errors: responses.filter((r) => r.error),
-    };
+    } as any;
   }
 
   /**
    * Get latency metrics
    */
   public async getLatencyMetrics({ range = 24, endDate = getEndDate(), params }: GetMetricsProps) {
-    const prange = getPRange(range);
-
-    // get p95 of requests in last 5min
-    const p95 = this.client.query({
-      query: `histogram_quantile(0.95, sum(rate(cosmo_router_http_request_duration_milliseconds_bucket{${this.getQueryLabels(
-        params,
-      )}}[${range}h])) by (le))`,
-    });
+    const granule = getGranularity(range);
+
+    const [start, end] = getDateRange(endDate, range);
+    const [prevStart, prevEnd] = getDateRange(endDate, range, range);
+
+    const queryLatency = (quantile: string, start: number, end: number) => {
+      return this.chClient.queryPromise<{ value: number }>(`
+        WITH
+          toDateTime('${start}') AS startDate,
+          toDateTime('${end}') AS endDate
+        SELECT quantile(0.5)(value) as value FROM (
```

Comment:
This looks incorrect. Building a media on top of the PX quantile is odd. We can just return the computed quantile `func_histogram_v2` for all operations.

Reply:
I wasn't getting correct value until I added this, this gets the median value of the p95 in the time range to my understanding.